Scrapper-Service/plugins/guide2research.py

```diff
@@ -0,0 +1,177 @@
+from typing import List, Dict
+
+import requests
+from bs4 import BeautifulSoup as bs
+
+import logging
+from commons import PageParsingError, Scrapper
+from datamodels import Conference, Metadata
+
+
+class Guide2ResearchScrapper(Scrapper):
+    def __init__(self, **config):
+        super().__init__(context_name=__name__, **config)
+        self.base_address = "http://www.guide2research.com/topconf/"
+        self.site_name = "guide2research"
+
+    def extract_and_push(self):
+        '''
+        Main calling method for g2r scraper
+        Collects the links of all top conferences available
+        and posts them to db
+        '''
+        conference_links = self._get_conferences_list()
+        for link in conference_links:
+            try:
+                conf_data = self._parse_conference(link=link)
+                self.push_todb(conf_data)
+            except Exception as e:
+                self.logger.error(
+                    f'Error while parsing page {link}, find full trace {e}')
+
+    def parse_action(self):
+        self.extract_and_push()
+
+    def _get_conferences_list(self) -> List[str]:
+        '''
+        parses topconferences baseurl and returns -
+        - a list of available links
+        '''
+        base_address = self.base_address
+        try:
+            page = requests.get(base_address)
+            if page.status_code == 200:
+                content = page.content
+        except Exception as e:
+            PageParsingError(
+                f"The following error occured when trying to parse a page {e}")
+        return self._parse_base(content)
+
+    def _parse_base(self, content: bytes) -> List[str]:
+        '''
+        Wrapper to parse the `/topconf` page and returns
+        a list of links for active conferences
+
+        Args:
+        content: bytes
+
+        Returns:
+        list of conferences: list
+        '''
+        soup = bs(content, 'html5lib')
+        div_el = soup.find('div', attrs={'id': 'ajax_content'})
+        conf_tables = div_el.findAll('table')
+        links = []
+        for table in conf_tables:
+            anchor = table.h4.a
+            if not anchor:
+                pass
+            else:
+                link = f"http://www.guide2research.com{anchor['href']}"
+                links.append(link)
+        return links
+
+    def _parse_conference(self, link: str) -> Conference:
+        '''
+        Parses individual conference page and,
+        return Conference object
+
+        Args:
+        ---
+        link: str
+
+        Returns:
+        ---
+        Conference: object
+        '''
+        try:
+            page = requests.get(link, allow_redirect=True)
+            if page.status_code == 200:
+                content = page.content
+        except Exception as e:
+            PageParsingError(
+                f'The following error occured while parsing a page {e}')
+        soup = bs(content, 'html5lib')
```

Comment:
The same case as I specified above. Here if we come across an Exception, or the page does not return 200, then the `content` variable would not be formed, thus would give an error.